test/distributed/fsdp/test_fsdp_optim_state.py

```diff
@@ -504,43 +539,71 @@ def test_scatter_full_optim_state_dict_nested(
     def test_scatter_full_optim_state_dict_transformer(self) -> None:
         """Tests :meth:`scatter_full_optim_state_dict` for an FSDP-root
         transformer model with shared parameters."""
-        self._test_shard_full_optim_state(
-            model_class="transformer", use_multiple_param_groups=False,
+        self._test_load_optim_state(
+            model_class=_ModelClass.TRANSFORMER,
+            use_multiple_param_groups=False,
             halve_world_size=True,
             osd_comm_method=_OSDCommMethod.SCATTER_FULL_OSD,
         )
 
-    def _test_shard_full_optim_state(
+    @skip_if_lt_x_gpu(2)
+    @parametrize("use_multiple_param_groups", [False, True])
+    def test_flatten_sharded_optim_state_dict_nested(
```

Comment:
Would I be correct to say that these added tests do not exercise resharding to different world size? They are just sanity checking that you can save and load a fixed sharded optim state dict under the same process group configuration?

Reply:
Resharding sharded_optim_state is not supported yet. So yes, these tests do not test resharding case. These tests test the optimizer values and keys are correct. It is redundant for `test_flatten_sharded_optim_state_dict_transformer()` to create two models as they are the same.  But `test_flatten_sharded_optim_state_dict_nested()` needs to create two models to test wrap alternative feature. Reusing `_test_load_optim_state` to avoid duplicating the test code.